source/slang/slang-parameter-binding.cpp

```diff
@@ -1042,15 +1041,64 @@ static void addExplicitParameterBindings_GLSL(
             return;
     }
 
-    // If we didn't find any matches, then bail
-    if(!resInfo)
+    // if we found resInfo, we add the explicit binding
+    if (resInfo)
+    {
+        auto kind = resInfo->kind;
+        auto count = resInfo->count;
+        semanticInfo.kind = kind;
+
+        addExplicitParameterBinding(context, parameterInfo, varDecl, semanticInfo, count);
+        return;
+    }
+
+    // See if we can infer vulkan binding from HLSL if we have such options set
+    auto hlslToVulkanLayoutOptions = context->getTargetRequest()->getHLSLToVulkanLayoutOptions();
+
+    if (!hlslToVulkanLayoutOptions)
+    {
+        _maybeDiagnoseMissingVulkanLayoutModifier(context, varDecl);
+        return;
+    }
+
+    // Do we have any vulkan shift settings
+    auto hlslRegSemantic = varDecl.getDecl()->findModifier<HLSLRegisterSemantic>();
+
+    if (hlslRegSemantic == nullptr)
+    {
+        // We don't have a HLSL binding, so we can't infer, so we can't assign an infered explicit binding
         return;
+    }
+            
+    // Get the HLSL binding info
+    const auto hlslInfo = ExtractLayoutSemanticInfo(context, hlslRegSemantic);
+    if (hlslInfo.kind != LayoutResourceKind::None)
+    {
+        // We need to map to the GLSL binding types
+        HLSLToVulkanLayoutOptions::Kind vulkanKind = HLSLToVulkanLayoutOptions::getKind(hlslInfo.kind);
+        if (vulkanKind != HLSLToVulkanLayoutOptions::Kind::Invalid)
+        {
+            const auto shift = hlslToVulkanLayoutOptions->getShift(vulkanKind, Index(hlslInfo.space));
+            if (shift != HLSLToVulkanLayoutOptions::kInvalidShift)
```

Comment:
If the shfit for this particular type of register isn't specified, do we want to issue a missing vulkan binding diagnostic instead of failing silently?